cpp/src/arrow/dataset/scanner_test.cc

```diff
@@ -125,8 +125,13 @@ TEST_F(TestSimpleScanner, ToTable) {
 
   auto scanner = std::make_shared<SimpleScanner>(sources, options_, ctx_);
   std::shared_ptr<Table> actual;
+
   ASSERT_OK(scanner->ToTable(&actual));
+  AssertTablesEqual(*expected, *actual);
 
+  // There is no guarantee on the ordering when using multiple threads, but
+  // since the RecordBatch is always the same it will pass.
```

Comment:
Maybe at some point we need AssertBatchSetsEqual or so

Reply:
gmock matchers provide a version I think, but I didn't want to refactor everything.